.github/workflows/auto-merge.yml

```diff
@@ -0,0 +1,22 @@
+name: Automerge PRs
+
+on:
+  schedule:
+    - cron: "*/10 * * * *" # Run every 10 minutes
+
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Automerge PRs with "automerge" label created within the last month and are mergeable
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          retries: 3
```

Comment:
Does this really work?

Reply:
I think what `retries` does is retrying on 400, 401, 403, 404, 422.

Reply:
Yes, we can customize the retryable status if we want
https://github.com/actions/github-script#retries

Reply:
Looks like we can only customize status codes.

Reply:
I understand your point, but do we need to retry for non-mergeable PRs in one workflow? I thought this case is handled by scheduling. Since we run the job every 10 minutes, temporarily non-meargeable jobs will be merged in the following cycles.

Reply:
Let's say we have two PRs to auto-merge. PR 1 is merged. PR 2's `mergeable` value is null until GitHub is done with recomputing mergeability. We should wait for a while until it becomes true or false. If true, we can merge, otherwise skip.

Reply:
I understand, so this is an expected waiting time. Added the retry logic for that case too, please take a look.